keras_nlp/utils/text_generation.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Text generation utilities."""
+
+import tensorflow as tf
+
+
+def _handle_end_token(next_token, end_token_received, end_token_id):
+    filtered_next_token = next_token * (1 - end_token_received)
+    end_token_received = tf.cast(
+        filtered_next_token == end_token_id, dtype=next_token.dtype
+    )
+    return filtered_next_token, end_token_received
+
+
+def generate_text_greedy(
+    get_next_token_probability_fn, seed, max_length, end_token_id=None
```

Comment:
We should probably find a more concise name for `get_next_token_probability_fn`. Maybe just `token_probability_fn`?

Reply:
do we want to include the "next" in the function name?

Reply:
I was just reading the API design doc the other day, and this seems to butt heads with recommendation there...

> Avoid OverlyLongAndSpecificNamingPatterns. If you find yourself with argument names with involve more than 3 subparts (e.g. “squared_operator_norm”), reconsider. Argument names should be intuitive and easy to remember.

I think brevity (even at the expense of fully describing this) might be good. So call it `token_probability_fn` so people can remember the name, and document as a function which gets the next token probability?

Reply:
We could go even shorter, say, `prediction_fn` if we want to add a `from_logits=True` arugment. And just document what it does clearly.

Reply:
sg!